src/ui/src/builder/useSocketTimeout.ts

```diff
@@ -16,23 +16,46 @@ export function useSocketTimeout(wf: Core, timeoutMin: number) {
 	const reconnecting = ref(false);
 	const prevent = ref(false);
 
+	const isMessagePending = computed(
+		() => wf.frontendMessageMap.value.size > 0,
+	);
+
+	const canCloseSocket = computed(
+		() => !prevent.value && !isMessagePending.value,
+	);
+
+	const abort = useAbortController();
+
 	function schedule() {
-		if (document.visibilityState === "visible") return;
+		if (!canCloseSocket.value || document.visibilityState !== "hidden")
+			return false;
```

Comment:
Do you need a return value?

Reply:
Yes, I'm using it for unit test 

https://github.com/writer/writer-framework/blob/655fc69622f39914e6c22d417f9b56eab6d10b97/src/ui/src/builder/useSocketTimeout.test.ts#L73-L73